backends/nats/nats.go

```diff
@@ -0,0 +1,486 @@
+	"github.com/go-logr/logr"
+	"github.com/nats-io/nats.go"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	// DefaultMaxReconnects - Max Reconnect attempts
+	DefaultMaxReconnects = 5
+	// DefaultReconnectWaitSeconds - Time to wait before attempting reconnect
+	DefaultReconnectWaitSeconds = 1
+	// DefaultNatsBucketDescription - Default description for bucket
+	DefaultNatsBucketDescription = "NATS bucket for simpleblob"
+	// DefaultNatsBucketReplicas - Default replicas for bucket
+	DefaultNatsBucketReplicas = 1
+	// DefaultMaxWaitSeconds - Default max wait for Jetstream replies
+	DefaultMaxWaitSeconds = 10
+)
+
+// internal enum to track auth type
+type authType int
+
+const (
+	username authType = iota
+	token
+	nkey
+	credentials
+	nkeyJwt
+	jwt
+	invalid
+)
+
+// Options enumerates various configuration options
+type Options struct {
+	// Params related to reconnection attempts
+	DisableRetryOnFailedConnect bool `yaml:"disableRetryOnFailedConnect"`
+	MaxReconnects               int  `yaml:"maxReconnects"`
+	ReconnectWaitSeconds        int  `yaml:"reconnectWait"`
+	MaxWaitSeconds              int  `yaml:"maxWaitSeconds"`
+	// Authentication options
+	NatsUsername            string `yaml:"natsUsername"`
+	NatsPassword            string `yaml:"natsPassword"`
+	NatsToken               string `yaml:"natsToken"`
+	NatsNkeySeedFilePath    string `yaml:"natsNkeySeedFilePath"`
+	NatsCredentialsFilePath string `yaml:"natsCredentialsFilePath"`
+	NatsJWTKeyFilePath      string `yaml:"natsJWTKeyFilePath"`
+	// TLS Options
+	// NatsTLSRootCA Specify a custom root CA file (e.g. for self-signed certs)
+	NatsTLSRootCA string `yaml:"natsTLSRootCA"`
+	// NatsTLSClientCert Specify a client cert file
+	NatsTLSClientCert string `yaml:"natsTLSClientCert"`
+	// NatsTLSClientKey Specify a client key file
+	NatsTLSClientKey string `yaml:"natsTLSClientKey"`
+	// Connection options
+	NatsURL string `yaml:"natsURL"`
+	// Storage Options
+	// NatsBucket defines the bucket name
+	NatsBucket string `yaml:"natsBucket"`
+	// NatsBucketReplicas defines number of replicas
+	NatsBucketReplicas int `yaml:"natsBucketReplicas"`
+	// NatsBucketDescription description for NatsBucket
+	NatsBucketDescription string `yaml:"natsBucketDescription"`
+	// CreateBucket tells us to try to create the bucket
+	CreateBucket bool `yaml:"create_bucket"`
+	// CreateBucketPlacementCluster optionally set the cluster placement value when creating a bucket
+	CreateBucketPlacementCluster string `yaml:"createBucketPlacementCluster"`
+	// CreateBucketPlacementTagList comma seperated list of optional placement tags when creating a bucket
+	CreateBucketPlacementTagList string `yaml:"createBucketTagList"`
+	// GlobalPrefix is a prefix applied to all operations, allowing work within a prefix
+	GlobalPrefix string `yaml:"global_prefix"`
+	// EncryptionKey -  key (hex format,256 bit) for encryption at rest
+	EncryptionKey string `yaml:"encryptionKey"`
+	// S3 Compatability
```

Comment:
Not S3.

Reply:
As per comment below, the problem was  the need for `tester/DoBackendTests` conformance.